state/meterstatus.go

```diff
@@ -76,13 +76,21 @@ type meterStatusDoc struct {
 	Info      string `bson:"info"`
 }
 
-// SetMeterStatus sets the meter status for the unit.
-func (u *Unit) SetMeterStatus(codeStr, info string) error {
+func isValidMeterStatusCode(codeStr string) (MeterStatusCode, error) {
 	code := MeterStatusFromString(codeStr)
 	switch code {
 	case MeterGreen, MeterAmber, MeterRed:
+		return code, nil
 	default:
-		return errors.Errorf("invalid meter status %q", code)
+		return MeterNotAvailable, errors.Errorf("invalid meter status %q", codeStr)
```

Comment:
I think I mentioned this in #7120... add a test (can be a mock) where this happens in the `juju status` cmd above.